src/components/products/viewProducts/ProductCard.js

```diff
@@ -1,56 +1,104 @@
 /* eslint-disable react/prop-types */
-import React from "react";
+import React, { useEffect, useState } from "react";
 import StarRating from "./StarRating.js";
 import Images from "./Images.js";
 import Categorize from "./Categorize.js";
 import SmallTable from "./SmallTable.js";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+
 import {
   formatDate,
   formatString,
   formatExpired,
 } from "../../../helpers/Format.js";
+import Button from "../../shared/Button.js";
+import { useDispatch, useSelector } from "react-redux";
 
 const ProductCard = ({ product, categories }) => {
-  console.log(product.average);
-  const { images } = product;
+  const { products,loading } = useSelector(
+    (state) => state.product,
+  )
+
+  const [isloading, setLoading]=useState(false);
+  const [product1,setProduct]=useState(product);
+  const token = localStorage.getItem("token");
+
+  const handleAvailability=()=>{
+    setLoading(true);
+    const axios = require('axios');
+
+  let config = {
+  method: 'patch',
+  maxBodyLength: Infinity,
+  url: `${process.env.BACKEND_URL}/products/9`,
```

Comment:
Why did you hard code the product id? Are you updating a static product?

Reply:
i forgot